### PR TITLE
Feature/ui gridster

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds angular-gridster in an aim for a better Grid system for dashboards.
- Pros
   - Improved performance.
   - Simpler code
   - Smoother drag-n-drop of widgets.
   - Resizing widgets
   - Metro-styled layout 
- Cons
    - [x] Need to improve resizing behavior (not part of gridster but something that we should do it) (https://github.com/caskdata/cdap/pull/2707)
    - [x] Should be backward compatible. (https://github.com/caskdata/cdap/commit/13bab849966f5042e535247b78c1888ead50f7e8)

##### NOTE: PR #2707 has to be merged before merging this PR to develop.

TODO:
   - Need to improve resizing to be more predictable.
   - Need to finesse resizing widgets.

End Goal:
  
dashboard

### CODE REVIEW

File: cdap-ui/app/directives/c3/charts.js [hunk 1]
```diff
@@ -113,6 +119,11 @@ ngC3.directive('c3Line', function () {
   return angular.extend({
     link: function (scope, elem, attr) {
       scope.initC3(elem, 'line', attr, {xtickcount: 5});
+      scope.clearChart = function() {
```
Comment (anwar6953) [inline cdap-ui/app/directives/c3/charts.js]: Seems like this function gets defined many times in this file. Can we simply put it in the `baseDirective`?

File: cdap-ui/app/features/dashboard/model/dashboards-model.js [hunk 2]
```diff
@@ -58,23 +61,16 @@ function (Widget, MyDataSource, mySettings, $q) {
...
   Dashboard.prototype.checkForEmptyDashboard = function() {
-    var isEmpty = true;
-    angular.forEach(this.columns, function(column) {
-      if (column.length > 0) {
-        isEmpty = false;
-      }
-    });
-    this.isEmpty = isEmpty;
+    this.isEmpty = this.columns.length === 0;
+    return this.columns.length === 0;
```
Comment (anwar6953) [inline cdap-ui/app/features/dashboard/model/dashboards-model.js]: return `this.isEmpty`

File: cdap-ui/app/directives/c3/charts.js (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/directives/c3/charts.js]: Cool. We don't need this function at all. Have removed this in this PR (https://github.com/caskdata/cdap/pull/2707).

File: cdap-ui/app/features/dashboard/model/dashboards-model.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/dashboard/model/dashboards-model.js]: Fixed this in this PR (https://github.com/caskdata/cdap/pull/2707) . This PR was supposed to be adding only angular-gridster. PR #2707  was supposed to be for re-organization  + cleanup.

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**